src/main/java/com/launchdarkly/openfeature/serverprovider/EvaluationContextConverter.java

```diff
@@ -87,10 +91,10 @@ private String getTargetingKey(String targetingKey, Value keyAsValue) {
 
         if (keyAsValue != null && keyAsValue.isString()) {
             // Targeting key takes precedence over key, because targeting key is in the spec.
-            targetingKey = !Objects.equals(targetingKey, "") ? targetingKey : keyAsValue.asString();
+            targetingKey = !isNullOrEmpty(targetingKey) ? targetingKey : keyAsValue.asString();
```

Comment:
The targeting key used to actually never be null, so a couple of these checks were not extensive enought.